### PR TITLE
Adjusted IOCTypes usage.

### PR DESCRIPTION
**Pull request by AWSHurneyt**
### Description
Adjusted IOCTypes.

The jar that includes the changes in PR https://github.com/opensearch-project/security-analytics-commons/pull/18 is included in this PR.

### Testing
The `SATIFSourceConfigRestApiIT` test suite requires a additional system parameter inputs to execute; `bucketName`, `region`, and `roleArn`. The github CI is **not** currently configured to supply those inputs. Those tests passed locally with those inputs provided.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by AWSHurneyt** [comment]
Security, and multi-node workflows are failing for unrelated reasons. They also failed in previous PRs.
https://github.com/opensearch-project/security-analytics/pull/1133/checks

Will investigate those failures separately.

File: src/main/java/org/opensearch/securityanalytics/threatIntel/iocscan/service/SaIoCScanService.java [hunk 1]
```diff
@@ -375,9 +376,9 @@ private static SearchRequest getSearchRequestForIocType(List<String> indices, St
         SearchRequest searchRequest = new SearchRequest(indices.toArray(new String[0]));
         BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery();
         // add the iocs sublist
-        boolQueryBuilder.must(new TermsQueryBuilder(STIX2.VALUE_FIELD, iocsSublist));
+        boolQueryBuilder.must(new TermsQueryBuilder(STIX2.VALUE_FIELD + ".keyword", iocsSublist));
         // add ioc type filter
-        boolQueryBuilder.must(new TermsQueryBuilder(STIX2.TYPE_FIELD, iocType.toLowerCase()));
+        boolQueryBuilder.must(new TermsQueryBuilder(STIX2.TYPE_FIELD + ".keyword", iocType.toLowerCase(Locale.ROOT)));
```
Comment (AWSHurneyt) [inline src/main/java/org/opensearch/securityanalytics/threatIntel/iocscan/service/SaIoCScanService.java:381]: For context, after changing the IOCTypes from using underscores (e.g., `domain_name`) to using hyphens (e.g., `domain-name`), it was necessary to append the fields with `.keyword` in order for this search request to return results. This change is needed to ensure the potentially-hyphenated values of the fields are being queried as-is.

Without these changes, the query returns 0 results despite a `match_all` query confirming there are docs in the concrete index. This was the behavior regardless of whether the index mappings were specified during index creation.

File: src/main/java/org/opensearch/securityanalytics/model/DetailedSTIX2IOCDto.java [hunk 2]
```diff
@@ -177,7 +176,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder.startObject()
                 .field(STIX2IOC.ID_FIELD, ioc.getId())
                 .field(STIX2IOC.NAME_FIELD, ioc.getName())
-                .field(STIX2IOC.TYPE_FIELD, ioc.getType())
+                .field(STIX2IOC.TYPE_FIELD, ioc.getType().getType())
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/model/DetailedSTIX2IOCDto.java:179]: why is it not ioc.getType().toString()??

Comment (AWSHurneyt) [reply src/main/java/org/opensearch/securityanalytics/model/DetailedSTIX2IOCDto.java]: Revised.

**Review by eirsep — APPROVED** [review]

**Review by jowg-amazon — APPROVED** [review]

**Pull request merged**